lib/tasks/fix_action_text.rake

```diff
@@ -0,0 +1,134 @@
+# Usage:
+#   bundle exec rails "maintenance:fix_action_text_sgids[OLD_SECRET_KEY_BASE]"
+#
+# Arguments:
+#   OLD_SECRET_KEY_BASE (optional): The previous secret_key_base. If provided, the task
+#                                   will deterministically decode old SGIDs to find the
+#                                   correct blob. If omitted, it falls back to a heuristic
+#                                   matching filename and filesize.
+#
+# related:
+# https://github.com/rails/rails/pull/39623
+# https://github.com/rails/rails/issues/40435#issuecomment-903398832
+# https://discuss.rubyonrails.org/t/how-to-rotate-secrek-key-base-without-breaking-activestorage-actiontext-attachments/80865
+namespace :maintenance do
+  desc "Regenerate ActionText SGIDs for ActiveStorage attachments after secret_key_base change"
+  task :fix_action_text_sgids, [:old_secret_key_base] => :environment do |_, args|
+    require "nokogiri"
+    Rails.logger.debug("Starting ActionText SGID repair...")
+
+    host = ENV.fetch("URL_HOST")
+    protocol = Rails.env.production? ? "https" : "http"
+    url_options = { host: host, protocol: protocol }
```

Comment:
I'm not sure I understand how you want to use this script in production. You will _not_ connect to it from outside, but execute this in the Rails console, right? Then something like `localhost` should be used in this scenario as well for it to work. Otherwise `URL_HOST` points to `vignettes.mampf.media`, which will probably not work as the Rails console knows nothing about it.

Reply:
No, this is a rake task that will be executed from a shell inside the container via
`bundle exec rails "maintenance:fix_action_text_sgids[OLD_SECRET_KEY_BASE]"`. I would assume that the `ENV` variables are present there. It is important that `URL_HOST` points to `vignettes.mampf.media` as the links pointing to the old server will need to be replaced in the database.